src/components/select/SelectReducers.ts

```diff
@@ -0,0 +1,67 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {ListBoxActions} from '../listBox/ListBoxActions';
+import {ISelectPayload, SelectActions} from './SelectActions';
+
+export interface ISelectState {
+  id: string;
+  open: boolean;
+}
+
+export const selectInitialState: ISelectState = {id: undefined, open: false};
+export const selectCompositeInitialState: ISelectState[] = [];
+
+export const selectReducer = (state: ISelectState = selectInitialState, action: IReduxAction<ISelectPayload>): ISelectState => {
+  switch (action.type) {
+    case SelectActions.add:
+      return {
+        id: action.payload.id,
+        open: state.open,
+      };
+    case SelectActions.close:
+    case SelectActions.open:
+    case SelectActions.toggle:
```

Comment:
you could move this if logic at the top like my proposition in another reducer file